### DISCUSSION TITLE
Flexible Content Item will be removed if there are no attributes set.

### DISCUSSION THREAD

**Discussion by thoresuenert** [whitecube/nova-flexible-content#124] [General]
We have a use case where we want to map a flexible content block to a static template, in other words: There no Fields for that layout.

What is the reason behinde the `!$attributes` check in the following line:
https://github.com/whitecube/nova-flexible-content/blob/414de90aa9356d4733b23593b15b6d5e7724d03f/src/Concerns/HasFlexible.php#L121

This leads to removal of the flexible content blocks with empty attributes `attributes: []`.

When we remove the attributes check everything is fine, but i dont know if this introduce a bug somewhere else.

Code Example:
```php
class MenuLayout extends Layout
{
    /**
     * The layout's unique identifier
     *
     * @var string
     */
    protected $name = 'module.menu';

    /**
     * The displayed title
     *
     * @var string
     */
    protected $title = 'Menu';

    /**
     * Get the fields displayed by the layout.
     *
     * @return array
     */
    public function fields()
    {
        return [
        ];
    }
}
```

```php
class Module extends Model
{
    use HasFlexible;

    public $guarded = [];

    protected $casts = [
        'content' => 'array',
    ];

    public function flexibleContent()
    {
// ModulePreset::mapping() will return an array ['name' => Layout::class]
// the MenuLayout is registered in the preset
        return $this->flexible('content', ModulePreset::mapping()->toArray());
    }
}
```

The database stores the following string for content:
`[{"key": "cddb22bae645ec22", "layout": "module.menu", "attributes": [] }]`

The following code does not render the view as aspected because `->flexibleContent()` is empty:
```php
@foreach($module->flexibleContent() as $content)
    @include($content->name(), ['module' => $content])
@endforeach
```

**Comment by toonvandenbos** [comment]
Hi @thoresuenert,

Did you try using a custom resolver instead of providing the flexible's value by hand (in `$this->flexible('content', ModulePreset::mapping()->toArray());`)?

I'm not sure what is going on here, maybe I do not understand your use case right.
